### PR TITLE
Remove Needless Forking in GatewayService

### PR DESCRIPTION
**Pull request by original-brownbear**
Submitting a cluster state update is cheap => no need to fork to generic here.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/gateway/GatewayService.java [hunk 1]
```diff
@@ -221,11 +214,7 @@ public ClusterState execute(final ClusterState currentState) {
                 logger.debug("cluster is already recovered");
                 return currentState;
             }
-
-            return Function.<ClusterState>identity()
-                .andThen(ClusterStateUpdaters::updateRoutingTable)
-                .andThen(ClusterStateUpdaters::removeStateNotRecoveredBlock)
-                .apply(currentState);
+            return ClusterStateUpdaters.removeStateNotRecoveredBlock(ClusterStateUpdaters.updateRoutingTable(currentState));
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/gateway/GatewayService.java:217]: Unrelated but this was just the weirdest way ever to call two functions ...

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/gateway/GatewayService.java:217]: It's pretty weird that this is even two functions, this is the only place they're called (apart from `ClusterStateUpdatersTests` which really needn't test the two things in isolation either). I've nearly opened a PR to fix this a number of times myself, thanks.

**Review by DaveCTurner — APPROVED** [review]
LGT<

**Comment by original-brownbear** [comment]
Thanks David!

**Pull request merged**